src/parse.d

```diff
@@ -2851,13 +2853,15 @@ public:
                 constraint = parseConstraint();
             }
         }
+
         switch (tok)
         {
         case TOKclass:
         case TOKinterface:
             {
                 if (!id)
-                    error("anonymous classes not allowed");
+                    error(loc, "anonymous classes not allowed");
```

Comment:
This is a small diagnostic improvement. Without the `loc`, this error message will show the location of *current* token that is immediately after the `class` keyword.